File: 135022193-main/2023/Python CS50/sentimental-readability/readability.py
from cs50 import get_string


def main():
    letter_count = 0
    words_count = 1
    sentences_count = 0
    text = get_string("Text: ")
    letter_count = count_letters(text, letter_count)
    words_count = count_words(text, words_count)
    sentences_count = count_sentences(text, sentences_count)
    print_grade(letter_count, words_count, sentences_count)


def count_letters(text, letter_count):
    for i in range(len(text)):
        if text[i].isalpha() == True:
            letter_count += 1
    return letter_count


def count_words(text, words_count):
    for i in range(len(text)):
        if text[i] == " ":
            words_count += 1
    return words_count


def count_sentences(text, sentences_count):
    for i in range(len(text)):
        if text[i] in [".", "!", "?"]:
            sentences_count += 1
    return sentences_count


def print_grade(letter_count, words_count, sentences_count):
    L = letter_count / words_count * 100
    S = sentences_count / words_count * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    index = round(index)
    if index > 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


main()

File: 135022193-main/2024/Done/credit/readability/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main()
{
    string text = get_string("Text: ");
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);

    float L = letters / words * 100;
    float S = sentences / words * 100;

    float tmp = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(tmp);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", index);
    }
}

int count_letters(string text)
{
    int sum = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            sum++;
        }
    }
    return sum;
}

int count_words(string text)
{
    int sum = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            sum++;
        }
    }
    return sum;
}

int count_sentences(string text)
{
    int sum = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sum++;
        }
    }
    return sum;
}
